services/src/tasks/in_memory.rs

```diff
@@ -165,6 +212,99 @@ impl TaskManager<SimpleTaskManagerContext> for SimpleTaskManager {
 
         Ok(result)
     }
+
+    async fn abort(&self, task_id: TaskId, force: bool) -> Result<(), TaskError> {
+        let mut write_lock = self.write_lock_all().await;
+
+        let task_handle = write_lock.remove_task_handle(task_id)?;
```

Comment:
I think the task itself will not be removed if the there is an error between remove_task_handle() and cleanup

Reply:
[cac20e7](https://github.com/geo-engine/geoengine/pull/586/commits/cac20e79e8085587b8f05c3a00c362d816299ddb)

Reply:
Please take a look if this is better now.
Do we clean-up on error? I think so, but then I have to add another route again:
 * abort clean-up vs error clean-up

Reply:
[597dceb](https://github.com/geo-engine/geoengine/pull/586/commits/597dcebde596be5eb51da3d843167eb18a2c3603)

Reply:
can you add tests for these scenarios?

Reply:
[c5ffe4d](https://github.com/geo-engine/geoengine/pull/586/commits/c5ffe4d7f7f113ccd9e6cf446b07834caca25834)